src/cmdlib.sh

```diff
@@ -414,7 +414,11 @@ impl_rpmostree_compose() {
         # "cache2" has an explicit label so we can find it in qemu easily
         if [ ! -f "${workdir}"/cache/cache2.qcow2 ]; then
             qemu-img create -f qcow2 cache2.qcow2.tmp 10G
-            LIBGUESTFS_BACKEND=direct virt-format --filesystem=xfs --label=cosa-cache -a cache2.qcow2.tmp &&
+            if [ "$arch" = "ppc64le" ] ; then
+                source /usr/lib/coreos-assembler/libguestfish.sh "virt-format --filesystem=xfs --label=cosa-cache -a cache2.qcow2.tmp"
```

Comment:
Hmm, how does that work? What part of `libguestfish.sh` execute the argument we're passing here?

Could we just drop the conditional entirely and just do e.g.

```bash
(source /usr/lib/coreos-assembler/libguestfish.sh
 virt-format --filesystem=xfs --label=cosa-cache -a cache2.qcow2.tmp)
```

? Since that should work on all arches, right?

Reply:
This part for POWER: https://github.com/coreos/coreos-assembler/blob/ab0986f2104adbbee7e8e46f7e4ba207d29e19f3/src/libguestfish.sh#L18
To drop the conditional we would need to change the libguestfish to pass the arguments as well.
https://github.com/coreos/coreos-assembler/blob/ab0986f2104adbbee7e8e46f7e4ba207d29e19f3/src/libguestfish.sh#L9

Reply:
Sorry, I still don't follow here. :)

That `exec` is part of a `cat` which creates the wrapper for qemu, but AFAICT there's nothing actually running the arguments passed to `source` (which itself is kind of an unusual thing to do).

As a sanity-check, I did

```diff
diff --git a/src/libguestfish.sh b/src/libguestfish.sh
index be061c09..e722d2f0 100755
--- a/src/libguestfish.sh
+++ b/src/libguestfish.sh
@@ -10,7 +10,6 @@ export LIBGUESTFS_BACKEND=direct

 arch=$(uname -m)

-if [ "$arch" = "ppc64le" ] ; then
     tmp_qemu_wrapper=$(mktemp -tdp /tmp gf-vsmt.XXXXXX)
     qemu_wrapper=${tmp_qemu_wrapper}/qemu-wrapper.sh
        cat <<-'EOF' > "${qemu_wrapper}"
@@ -19,7 +18,6 @@ if [ "$arch" = "ppc64le" ] ; then
        EOF
     chmod +x "${qemu_wrapper}"
     export LIBGUESTFS_HV="${qemu_wrapper}"
-fi

 # http://libguestfs.org/guestfish.1.html#using-remote-control-robustly-from-shell-scripts
 GUESTFISH_PID=
```

And then in my shell:

```
$ source src/libguestfish.sh "echo foobar"
$
```

Reply:
>  To drop the conditional we would need to change the libguestfish to pass the arguments as well.

Can you expand on this? I'm not sure if we're talking about the same thing. What I mean is that, if we just source liguestfish.sh`, on all arches that already sets `export LIBGUESTFS_BACKEND=direct` which we want, and on ppc64le, will take care of creating the qemu wrapper. Then just calling `virt-format` separately should do the right thing always, right?